connectivity/FEATURE_BLE/source/generic/GapImpl.cpp

```diff
@@ -2469,6 +2447,93 @@ ble_error_t Gap::startAdvertising(
 }
 #endif
 
+#if BLE_FEATURE_EXTENDED_ADVERTISING
+void Gap::start_advertising_enable(
+    advertising_handle_t handle,
+    adv_duration_t maxDuration,
+    uint8_t maxEvents
+)
+{
+    /* remember the parameters that will be enabled when the last command completes */
+    if (_advertising_enable_queue.handle == ble::INVALID_ADVERTISING_HANDLE) {
+        /* special case when there is only one pending */
+        _advertising_enable_queue.handle = handle;
+        _advertising_enable_queue.max_duration = maxDuration;
+        _advertising_enable_queue.max_events = maxEvents;
+        _advertising_enable_queue.next = nullptr;
+    } else {
+        AdvertisingEnableStackNode_t** next = &_advertising_enable_queue.next;
+        /* move down the queue until we're over a nullptr */
+        if (*next) {
+            while ((*next)->next) {
+                next = &((*next)->next);
+            }
+            next = &(*next)->next;
+        }
+        *next = new AdvertisingEnableStackNode_t();
+        if (!*next) {
+            tr_error("Out of memory creating pending advertising enable node for handle %d", handle);
+            return;
+        }
+        (*next)->handle = handle;
+        (*next)->max_duration = maxDuration;
+        (*next)->max_events = maxEvents;
+        (*next)->next = nullptr;
+    }
+}
+
+void Gap::evaluate_advertising_enable_queue()
```

Comment:
I suggest `process` instead of `evaluate`. As it does an action and not just look at the queue.

Reply:
will rename